installer/pyinstaller/build-linux.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+set -eu
+binary_zip_filename=$1
+python_library_zip_filename=$2
+python_version=$3
+
+if [ "$python_library_zip_filename" = "" ]; then
+    python_library_zip_filename="python-libraries.zip";
+fi
+
+if [ "$python_version" = "" ]; then
+    python_version="3.7.9";
+fi
+
+yum install -y zlib-devel openssl-devel
+
+echo "Making Folders"
+mkdir -p .build/src
+mkdir -p .build/output/aws-sam-cli-src
+mkdir -p .build/output/python-libraries
+mkdir -p .build/output/pyinstaller-output
+cd .build
+
+echo "Copying Source"
+cp -r ../[!.]* ./src
+cp -r ./src/* ./output/aws-sam-cli-src
+
+echo "Installing Python"
+curl "https://www.python.org/ftp/python/${python_version}/Python-${python_version}.tgz" --output python.tgz
+tar -xzf python.tgz
+cd Python-$python_version
+./configure --enable-shared
+make -j8
+make install
+cd ..
+
+echo "Installing Python Libraries"
+python3 -m venv venv
+./venv/bin/pip install -r src/requirements/reproducible-linux.txt
+
+echo "Copying All Python Libraries"
+cp -r ./venv/lib/python*/site-packages/* ./output/python-libraries
+
+echo "Installing PyInstaller"
+./venv/bin/pip install -r src/requirements/pyinstaller-build.txt
+
+echo "Building Binary"
+cd src
+../venv/bin/python -m PyInstaller -D --clean installer/pyinstaller/samcli.spec
+
+
+mkdir pyinstaller-output
+
+mv dist/sam pyinstaller-output/dist
+cp installer/assets/* pyinstaller-output
+cp /usr/local/lib/libcrypt.so.2 pyinstaller-output/dist/libcrypt.so.2
```

Comment:
If possible would be great to have some comments explaining why this are done the way they are.